### PR TITLE
replate `time::Duration` with `chrono::Duration` and add `max_age_time` method

### PR DESCRIPTION
**Pull request by Kilerd**
as #785 says, I replate it and add two more tests for cookies' max-age

### CODE REVIEW

**Comment by fafhrd91** [comment]
i'd like to replace time in `actix-http::cookie` as well. do you think you can do that?

**Comment by Kilerd** [comment]
sure, pls wait for a second

**Comment by Kilerd** [comment]
`actix-http` also use `time::Tm` to represent the time when the cookie expires, I think I need more time to replace it.

**Comment by Kilerd** [comment]
but looking through the code, I don't think we need to change the code using `Tm`

**Comment by DoumanAsh** [comment]
If we're going to use `chrono` we need to use it for parsing too.
There is no benefit in having `chrono::Duration` while retaining `time`

In addition to that I think `std::time::Duration` should `chono::Duration`

**Comment by Kilerd** [comment]
it means we need to maintain another copy of cookies repo, and I also do not know why `actix-http` need to reimplement cookies(compare these two repos, I can't see a lot of differences) rather than using `cookie`.

**Comment by DoumanAsh** [comment]
@Kilerd we need own `cookie` because the existing one is no longer being maintained and in future, due to `ring` policy, it will be no longer possible to properly compile it as `ring 0.13.x` will be yanked completely.
We already have it so there is nothing bad about modifying it, since old `cookie` is unlikely to get updated anyway

**Comment by Dowwie** [comment]
@Kilerd What is wrong with using std::time::Duration?  What is the benefit of this change?

**Comment by Dowwie** [comment]
> i'd like to replace time in `actix-http::cookie` as well. do you think you can do that?

what is wrong with std::time crate?

**Comment by DoumanAsh** [comment]
@Dowwie we need parsing of time strings

**Comment by Dowwie** [comment]
> @Dowwie we need parsing of time strings

ah.  understood.  thanks

File: actix-http/src/cookie/builder.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 use std::borrow::Cow;
 
-use time::{Duration, Tm};
+use time::Tm;
```
Comment (fafhrd91) [inline actix-http/src/cookie/builder.rs:3]: we should drop `time`, there is no reason to keep both.

**Comment by fafhrd91** [comment]
@Kilerd could you also update actix-session/src/cookie.rs same as you did for identity

**Review by fafhrd91 — APPROVED** [review]

**Comment by fafhrd91** [comment]
thanks

**Comment by fafhrd91** [comment]
i had to remove chrono.

reasons:
* it can not parse rfc822 format
* it requires custom format string for rfc822
* internally it uses time crate

there is no reason to use it. i kept max_age changes though

**Comment by Kilerd** [comment]
sorry for the late responding

> it can not parse rfc822 format

yep when I try to use chrono parsing the cookie expire time, I also found this issue.

**Comment by Kilerd** [comment]
and why not just expose the `time:Duration` as public?

**Comment by fafhrd91** [comment]
it would introduce confusion and it is used only for cookies.

**Pull request merged**